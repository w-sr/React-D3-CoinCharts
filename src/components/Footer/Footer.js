import React from "react";
import styled from "styled-components";

import coinbaseLogo from "./assets/coinbase-logo.svg";
import { color, fontFamily, fontSize, fontWeight, size } from "../../styles/constants";

const StyledFooter = styled.div`
  color: ${color.white};
  font-family: ${fontFamily.regular};
  font-size: ${fontSize.medium};
  font-weight: ${fontWeight.medium};
  height: 0;
  line-height: 4em;
  text-align: center;
`;

const Image = styled.img`
  height: ${size.large};
`;

const Footer = props => (
  <StyledFooter {...props}>
    <span>Powered by </span>
    <a href="https://developers.coinbase.com/api/v2" target="_blank" rel="noopener noreferrer">
      <Image src={coinbaseLogo} alt="coinbase-logo" />
    </a>
  </StyledFooter>
);

export default Footer;
