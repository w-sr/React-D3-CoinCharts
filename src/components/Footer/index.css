.Footer {
  color: white;
  font-size: 16px;
  font-weight: 500;
  margin: 1.5em auto 0;
  text-align: center;
}

.Footer .logo {
  height: 16px;
}
