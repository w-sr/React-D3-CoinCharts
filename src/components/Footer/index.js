import React from 'react';

import logo from './assets/logo.svg';
import './index.css';

const Footer = () => (
  <div className="Footer">
    <span>Powered by </span>
    <a
      href="https://developers.coinbase.com/api/v2"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src={logo}
        className="logo"
        alt="coinbase"
      />
    </a>
  </div>
);

export default Footer;
