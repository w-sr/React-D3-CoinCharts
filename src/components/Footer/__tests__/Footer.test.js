import React from "react";
import { render } from "react-testing-library";

import Footer from "../";

describe("<Footer />", () => {
  it("renders without crashing", () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders logo w/ correct alt-text", () => {
    const { getByAltText } = render(<Footer />);
    const footerLogo = getByAltText("coinbase-logo");
    expect(footerLogo).not.toBeNull();
  });
});
