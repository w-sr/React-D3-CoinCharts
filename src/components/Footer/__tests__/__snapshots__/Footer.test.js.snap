// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders without crashing 1`] = `
<div
  class="sc-bdVaJa gqvhCj"
>
  <span>
    Powered by 
  </span>
  <a
    href="https://developers.coinbase.com/api/v2"
    target="_blank"
    rel="noopener noreferrer"
  >
    <img
      class="sc-bwzfXH cJoRuk"
      src="coinbase-logo.svg"
      alt="coinbase-logo"
    />
  </a>
</div>
`;
